EIPS/eip-7928.md

```diff
@@ -79,76 +81,81 @@ class AccountBalanceDiff(Container):
 
 BalanceDiffs = List[AccountBalanceDiff, MAX_ACCOUNTS]
 
-# Post-transaction nonce structures
+# Code Diff structures
+class CodeChange(Container):
+    tx_index: TxIndex
+    new_code: CodeData
+
+class AccountCodeDiff(Container):
+    address: Address
+    changes: List[CodeChange, MAX_TXS]
```

Comment:
Hmm, since code thus is immutable once it is deposited after the tx (due to EIP-6780) I think we do not need the `CodeChange` structure. We can rather just dump the tx index and the address in and we should be done here.

Reply:
Ahh wait this is likely due to 7702. Have to think a bit about this. I think if we include the nonce diffs then we should also be fine. So if a 7702-delegated EOA executes a CREATE, this will bump the EOAs nonce. We have to track this in the BAL otherwise we could end up in wrong execution paths. This could lead to txs included in the block being invalid (wrong nonce), but it could also be that we try to delegate the EOA using a 7702 transaction and instead check against the wrong nonce (so we either set the code or not).